### PR TITLE
PLT-423 dont create new environment ticket when environment already exists

### PR DESCRIPTION
**Pull request by tomaszCarton**

### CODE REVIEW

File: jira-environment-ticket-create/lib/index.js
Comment (tomaszCarton) [inline jira-environment-ticket-create/lib/index.js:1]: Autogenerated, not sure why it wasn't here before

Comment (aarontwf) [reply jira-environment-ticket-create/lib/index.js:1]: Has something changed the directory from `dist` to `lib`?

File: jira-environment-ticket-create/src/index.ts [hunk 1]
```diff
@@ -12,8 +12,29 @@ async function run() {
...
     const nameField = core.getInput('nameField', { required: true });
     const urlField = core.getInput('urlField', { required: true });
+    const projectKey = core.getInput('projectKey', { required: true });
+    const jiraEnvironmentField = core.getInput('jiraEnvironmentField', { required: true });
 
     const jiraBase64Credentials = Buffer.from(`${jiraUsername}:${jiraPassword}`).toString('base64');
+    const environmentJql = `project = ${projectKey} AND "${jiraEnvironmentField}" ~ "${environmentName}"`;
+
+    core.info('Checking if issue already exists');
+    const existingUrl = encodeURI(`https://${jiraServer}/rest/api/latest/search?jql=${environmentJql}&fields=labels`);
```
Comment (aarontwf) [inline jira-environment-ticket-create/src/index.ts]: Guessing this isn't needed?
```suggestion
    const existingUrl = encodeURI(`https://${jiraServer}/rest/api/latest/search?jql=${environmentJql}`);
```

File: jira-environment-ticket-create/src/index.ts [hunk 2]
```diff
@@ -12,8 +12,29 @@ async function run() {
...
+        'Authorization': `Basic ${jiraBase64Credentials}`,
+        'Content-Type': 'application/json',
+      },
+    });
+
+    const matchingIssues: any = await existingResponse.json();
+
+    if (matchingIssues.total !== 0) {
+      core.warning(`A ticket for environment ${environmentName} already exists.`);
+      return;
```
Comment (aarontwf) [inline jira-environment-ticket-create/src/index.ts:36]: Should this be a `core.setFailed`?

File: jira-environment-ticket-create/lib/index.js
Comment (blake-cartoncloud) [reply jira-environment-ticket-create/lib/index.js:1]: Delete it and rerun `npm run build` see if it returns

File: jira-environment-ticket-create/action.yml [hunk 3]
```diff
@@ -19,6 +19,9 @@ inputs:
   projectId:
     description: 'Jira project id to create ticket in'
     required: true
+  projectKey:
+    description: 'Jira project key for checking if same name issue doesnt exit'
```
Comment (blake-cartoncloud) [inline jira-environment-ticket-create/action.yml]:
```suggestion
    description: 'Jira project key for checking if issue with same name does not exist'
```

File: jira-environment-ticket-create/action.yml [hunk 4]
```diff
@@ -28,6 +31,10 @@ inputs:
   urlField:
     description: 'JSON field for the environment URL'
     required: true
+  jiraEnvironmentField:
+    description: 'JQL field used to find the given environment'
+    required: false
+    default: 'Environment Key[Short text]'
```
Comment (blake-cartoncloud) [inline jira-environment-ticket-create/action.yml:37]: Strange default?

**Review by blake-cartoncloud — APPROVED** [review]

File: jira-environment-ticket-create/src/index.ts (hunk 2 above)
Comment (tomaszCarton) [reply jira-environment-ticket-create/src/index.ts:36]: No, we don't want to fail the action. Running create for the second time agains the environment should not cause any harm.

File: jira-environment-ticket-create/lib/index.js
Comment (tomaszCarton) [reply jira-environment-ticket-create/lib/index.js:1]: Yeah, it recreates it every time, some other actions also have the /lib dir autogenerated.
Will just leave it this way. We can have a separate ticket to clean all of them if we decide it is worth.

File: jira-environment-ticket-create/src/index.ts (hunk 2 above)
Comment (aarontwf) [reply jira-environment-ticket-create/src/index.ts:36]: Ah, I thought you'd want it to so whoever triggered the action can tell they tried to create an environment someone else is using without digging through the logs to see this warning.

**Review by aarontwf — APPROVED** [review]

File: jira-environment-ticket-create/lib/index.js
Comment (blake-cartoncloud) [reply jira-environment-ticket-create/lib/index.js:1]: Must be `tsc` generating lib and esbuild generating `dist`. Doubt you need the tsc build

File: jira-environment-ticket-create/action.yml (hunk 4 above)
Comment (tomaszCarton) [reply jira-environment-ticket-create/action.yml:37]: Yeah, whats stranger is it doesn't get overwritten :D
Took the piece of code for checking the ticket from jira-environment-ticket-delete and and keeping it the same here.

**Pull request merged**